Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/LocationUpdateService.java

```diff
@@ -368,8 +371,152 @@ private void updateDriverStatus(Boolean status) {
         });
     }
 
+    private void getRoute(Double startLat, Double startLng, Double endLat, Double endLng) {
+        StringBuilder result = new StringBuilder();
+
+        SharedPreferences sharedPref = context.getSharedPreferences(context.getString(R.string.preference_file_key), Context.MODE_PRIVATE);
+
+        String regToken = sharedPref.getString("REGISTERATION_TOKEN", "null");
+        String baseUrl = sharedPref.getString("BASE_URL", "null");
+        String version = sharedPref.getString("VERSION_NAME", "null");
+        try {
+            String url = baseUrl + "/trip/route";
+            HttpURLConnection connection = (HttpURLConnection) (new URL(url).openConnection());
+            connection.setRequestMethod("POST");
+            connection.setRequestProperty("Content-Type", "application/json");
+            connection.setRequestProperty("token", regToken);
+            connection.setRequestProperty("x-client-version", version);
+
+            JSONObject payload = new JSONObject();
+
+            JSONObject startLatLng = new JSONObject();
```

Comment:
Can we create a common function for this?
```
            JSONObject startLatLng = new JSONObject();
            startLatLng.put("lat", startLat);
            startLatLng.put("lon", startLng);

            JSONObject endLatLng = new JSONObject();
            endLatLng.put("lat", endLat);
            endLatLng.put("lon", endLng);
```
Something like this

```
getLatLongObject (lat, long) {
           JSONObject latLng = new JSONObject();
            latLng.put("lat", endLat);
            latLng.put("lon", endLng);
return latLng
}
```

Reply:
Done